### PR TITLE
Rover: Make the definition name a class type

### PR DESCRIPTION
**Pull request by muramura**
Definition names were prefixed with group names. I can make definition names clearer by making the group name the class name.

### CODE REVIEW

File: Rover/defines.h [hunk 1]
```diff
@@ -88,11 +88,11 @@ enum fs_ekf_action {
 
 #define DISTANCE_HOME_MINCHANGE 0.5f  // minimum distance to adjust home location
 
-enum pilot_steer_type_t {
-    PILOT_STEER_TYPE_DEFAULT = 0,
-    PILOT_STEER_TYPE_TWO_PADDLES = 1,
-    PILOT_STEER_TYPE_DIR_REVERSED_WHEN_REVERSING = 2,
-    PILOT_STEER_TYPE_DIR_UNCHANGED_WHEN_REVERSING = 3,
+enum class pilot_steer_type : int8_t {
```
Comment (peterbarker) [inline Rover/defines.h:91]:
```suggestion
enum class PilotSteerType : int8_t {
```

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: Rover/defines.h (hunk 1 above)
Comment (muramura) [reply Rover/defines.h]: @peterbarker san.
I agree. I will change it to PilotSteerType.

File: Rover/defines.h [hunk 2]
```diff
@@ -88,11 +88,11 @@ enum fs_ekf_action {
 
 #define DISTANCE_HOME_MINCHANGE 0.5f  // minimum distance to adjust home location
 
-enum pilot_steer_type_t {
-    PILOT_STEER_TYPE_DEFAULT = 0,
-    PILOT_STEER_TYPE_TWO_PADDLES = 1,
-    PILOT_STEER_TYPE_DIR_REVERSED_WHEN_REVERSING = 2,
-    PILOT_STEER_TYPE_DIR_UNCHANGED_WHEN_REVERSING = 3,
+enum class PilotSteerType : int8_t {
```
Comment (peterbarker) [inline Rover/defines.h:91]:
```suggestion
enum class PilotSteerType : uint8_t {
```

File: Rover/mode.cpp [hunk 3]
```diff
@@ -123,8 +123,8 @@ void Mode::get_pilot_desired_steering_and_throttle(float &steering_out, float &t
     // check for special case of input and output throttle being in opposite directions
     float throttle_out_limited = g2.motors.get_slew_limited_throttle(throttle_out, rover.G_Dt);
     if ((is_negative(throttle_out) != is_negative(throttle_out_limited)) &&
-        ((g.pilot_steer_type == PILOT_STEER_TYPE_DEFAULT) ||
```
Comment (peterbarker) [inline Rover/mode.cpp:126]: This would be a *lot* neater if you made g.pilot_steer_type `AP_Enum<PilotSteerType>`

**Review by peterbarker — CHANGES_REQUESTED** [review]
Has to compile!

**Comment by peterbarker** [comment]
I've made the changes I suggested in my review.

**Comment by peterbarker** [comment]
Compiler output is zero-bytes or no-compiler-output.

Tested lightly in SITL by frobbing the PILOT_STEER_TYPE parameter

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**